src/hats/io/file_io/file_io.py

```diff
@@ -213,21 +212,18 @@ def write_parquet_metadata(
     )
 
 
-def read_fits_image(map_file_pointer: str | Path | UPath):
+def read_fits_image(map_file_pointer: str | Path | UPath) -> np.ndarray:
     """Read the object spatial distribution information from a healpix FITS file.
 
     Args:
-        file_pointer: location of file to be written
+        map_file_pointer (path-like): location of file to be written
+
     Returns:
         one-dimensional numpy array of integers where the
         value at each index corresponds to the number of objects found at the healpix pixel.
     """
     map_file_pointer = get_upath(map_file_pointer)
-    with tempfile.NamedTemporaryFile() as _tmp_file:
-        with map_file_pointer.open("rb") as _map_file:
-            map_data = _map_file.read()
-            _tmp_file.write(map_data)
-            return hp.read_map(_tmp_file.name, nest=True)
+    return Skymap.from_fits(map_file_pointer).values
```

Comment:
Really? They support reading through fsspec? The weird temp file thing here is due to limitations with reading from remote systems. Did you test this change out with hats-cloudtests?

Reply:
Unfortunately you're right, I had not played with this on the cloud! Added a test to hats-cloudtests to make sure this still works as expected: https://github.com/astronomy-commons/hats-cloudtests/pull/46.